### PR TITLE
Add migration page

### PR DESCRIPTION
**Pull request by benjgil**
Add migration page with basic steps for migrating over legacy components to Harmony.
https://app.clickup.com/t/7r00c3

(will need to write more detailed steps at a later stage)

### CODE REVIEW

**Review by itaymendel — CHANGES_REQUESTED** [review]
Do not submit PRs for `dep-resolver` here.
submit a PR to this - https://github.com/teambit/bit/blob/master/scopes/dependencies/dependency-resolver/dependency-resolver.docs.md
and then @debs-obrien will simply `bit import` to update content in docs.

**Review by itaymendel — COMMENTED** [review]
I think that as a developer coming form Legacy to Harmony, if you'd describe what are the fundamental changes, it'll be easier for me to migrate my code. for example:

* **component structure** - Harmony is opinionated when it comes to structuring components in the file-system. So now each component *must* be in its own directory. so potentially you'll have to move some files around to fit that.
* **dependencies** - Harmony doesn't allow relative import statements between components. Each component gets a local module in `node_modules`, use them for all `import` statements in your code.
* **environments** - Harmony features a new approach for compilers/testers called *environments*. use the base environment per your component's stack (react, node, react-native) and customize it to fit any specific configuration requirement.
* **configuration** in harmony config is managed in a new file format - `workspace.jsonc` and not in your `package.json`.
* **module names** in harmony we removed the `@bit` prefix. component will be pre-fixed by `@<account name>`. so import statements should change.

i think that when these diffs are well explained, a dev should know how to fit their structure to the new setup.

**Review by itaymendel — COMMENTED** [review]
another important step here, and because legacy<>harmony are not compatible - you will have to tell the user to:

1. create a new repository - as you can only have one local `.git/bit` dir, and its contents will not be compatible. so for the migration period you can't use the same repo. what i recommend here is doing a fresh clone, run `bit ini --reset-hard` and only then migrate to harmony. afterwards create a new remote git-repo and push it there (this way you still get alll your git-history).
2. create a new scope. they are not compatible...

**Comment by benjgil** [comment]
Looks like I mixed up commits<>branches. Moving the deps resolver stuff to the relevant branch

**Comment by benjgil** [comment]
> Do not submit PRs for `dep-resolver` here.
> submit a PR to this - https://github.com/teambit/bit/blob/master/scopes/dependencies/dependency-resolver/dependency-resolver.docs.md
> and then @debs-obrien will simply `bit import` to update content in docs.

AFAIK that's already happened... The file for deps resolver on the Harmony branch already contains these edits, so no idea why this PR thinks these are changes :-?

File: docs/resources/migration.md
Comment (itaymendel) [inline docs/resources/migration.md]: upper case B (in Bit)

File: docs/resources/migration.md
Comment (itaymendel) [inline docs/resources/migration.md]: this is somewhat irrelevant and some of it actually goes against what you mentioned earlier.

also, i'd recommend setting up a new workspace and not reset previous one (again, not compatible, so what happens when i need to move backwards in time?)

I think you missed one of my comments from before, i actually noted about it

File: docs/resources/migration.md
Comment (itaymendel) [inline docs/resources/migration.md]: this is explained in a single bullet at the top, i don't see why you would want to repeat that?
unless you provide with techinical steps (in a future iteration?), skip it for now.

File: docs/resources/migration.md
Comment (itaymendel) [inline docs/resources/migration.md]: again, here you mention reseting the workspace on the same clone.

File: docs/resources/migration.md
Comment (itaymendel) [inline docs/resources/migration.md]: you are using the wrong link. just tell people that components should be dirs now and to use `bit add`. that;s it

File: docs/resources/migration.md
Comment (itaymendel) [inline docs/resources/migration.md]: no real reason for this. just link to workspace explenation and let them learn from it.

File: docs/resources/migration.md
Comment (itaymendel) [inline docs/resources/migration.md]: this is critical. very.
but you only mention it mid-doc and in a note.
this should be at the top.

File: docs/resources/migration.md
Comment (itaymendel) [inline docs/resources/migration.md]: there's a doc about it. no need to repeat it here. assume that who got here already read it.

File: docs/resources/migration.md
Comment (itaymendel) [inline docs/resources/migration.md]: i don't think these are the steps. perhaps this will be better:

1. do a fresh clone for your project
2. run `bit init --reset-hard`
3. run `bit init --harmony`
4. move files around and sort components in directories
5. configure `workspace.jsonc` (and link to the page)
5. run `bit add <component>` on a basic component
6. run `but start`
7. customize your env as needed + link
8. create a new scope
9. tag/export
10. link to CI-doc, as it is updated

File: docs/resources/migration.md
Comment (itaymendel) [inline docs/resources/migration.md]: missing explenation on BVM and the fact you can have both binaries running, so you can experiement with harmony while maintaining legacy

File: docs/resources/migration.md
Comment (itaymendel) [inline docs/resources/migration.md]: " and a repo aint big enough for the two of them"... these are docs, not blog post. we need to have concrete, technical explanations, not fluff...

File: docs/resources/migration.md
Comment (benjgil) [reply docs/resources/migration.md]: The technical steps are provided in the links  - the bullet point above is stating what will need to be done, this paragraph contains more concrete steps with links to the detailed instructions for carrying them out.
There really is no need to repeat the docs, but to reference them

File: docs/resources/migration.md
Comment (benjgil) [reply docs/resources/migration.md]: Yep, alongside creating a clone. It's up to them how they want to migrate, whether to get rid of their legacy bit or keep it alongisde

File: docs/resources/migration.md
Comment (benjgil) [reply docs/resources/migration.md]: 1) Not sure what's irrelevant about this. It's the exact list of steps needed to migrate. Care to elaborate?
2) What does it go against?
3) The workspace resetting / cloning is actually the first item in the list. Looks like I just left it in the middle though, so the second part (cloning and having it in a separate workspace) isnt there. But even if they clone, they still need to know how to reset the cloned repo as it will have legacy bit intertwined with it

In general this whole comment is full of general statements - please try to be more concrete in the feedback, it helps promote conversation and especially in understanding what the actual feedback is

File: docs/resources/migration.md
Comment (benjgil) [reply docs/resources/migration.md]: That's exactly what that link tells them to do...

File: docs/resources/migration.md
Comment (benjgil) [reply docs/resources/migration.md]: :?
The quoted section does link to the workspace explanation... (line 59)

File: docs/resources/migration.md
Comment (benjgil) [reply docs/resources/migration.md]: Moved up.
It's not relevant right at the top, will just confuse until we've explained the start of the process

File: docs/resources/migration.md
Comment (benjgil) [reply docs/resources/migration.md]: Really really dont assume that people who have gotten here understand all the benefits and are fully bent on migration. This might be where they give up unless we give them reasons to persist with the migration

File: docs/resources/migration.md
Comment (benjgil) [reply docs/resources/migration.md]: You're right, it's a better order. Will move environments to lower down

File: docs/resources/migration.md
Comment (benjgil) [reply docs/resources/migration.md]: docs need to be engaging, this isn't a blog post it's just informal language which can make the experience of working with a product more enjoyable. Little things.

File: docs/resources/migration.md
Comment (benjgil) [reply docs/resources/migration.md]: Right, will add

Comment (benjgil) [reply docs/resources/migration.md]: Done

File: docs/resources/migration.md
Comment (davidfirst) [inline docs/resources/migration.md]: @benjgil , there is an easier way to achieve this step by running `bit migrate --harmony`. See this: https://github.com/teambit/bit/issues/2795 for more details.

It was written a while ago, and wasn't tested since then, so you can ignore it altogether. Up to you.

**Review by itaymendel — APPROVED** [review]

**Comment by itaymendel** [comment]
@debs-obrien please merge this only after you merge #154 
otherwise you get broken links.

**Review by itaymendel — APPROVED** [review]

**Pull request merged**
